### PR TITLE
Merge main next release

### PR DESCRIPTION
**Pull request by timngyn**
#### Description of changes:
Merged in changes from here: https://github.com/aws-amplify/docs/commits/main/?since=2024-03-12&until=2024-03-15

Staging site: https://merge-main-next-release.d1ywzrxfkb9wgg.amplifyapp.com/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/components/Breadcrumbs/__tests__/Breadcrumbs.test.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+      expect(breadcrumbLink).toBeInTheDocument();
+      expect(breadcrumbLink).toHaveAttribute('href', route);
+    }
+  });
+
+  it('should replace "prev" with applicable version in Breadcrumbs text', async () => {
+    const component = (
+      <Breadcrumbs
+        route={'/gen1/javascript/prev/build-a-backend/auth/set-up-auth/'}
```
Comment (timngyn) [inline src/components/Breadcrumbs/__tests__/Breadcrumbs.test.tsx:48]: Had to update this to use `gen` route for the test to pass

File: src/components/Overview/__tests__/Overview.test.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    return {
+      query: { platform: 'react' }
+    };
+  }
+};
+
+jest.mock('next/router', () => routerMock);
+
+describe('Overview', () => {
+  const childPageNodes = getChildPageNodes('/gen1/[platform]/build-a-backend');
```
Comment (timngyn) [inline src/components/Overview/__tests__/Overview.test.tsx:18]: Had to update this to `gen1` route for test to pass

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/index.mdx
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/index.mdx:1]: Compare this to: https://github.com/aws-amplify/docs/blob/main/src/pages/gen2/build-a-backend/data/custom-business-logic/index.mdx

File: src/pages/[platform]/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx:1]: Compare this to https://github.com/aws-amplify/docs/blob/main/src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx

**Review by katiegoines — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Pull request merged**